src/clr-core/alert/alert.base.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import '@clr/core/button';
+import { CwcButton } from '@clr/core/button';
+import {
+  CommonStringsService,
+  event,
+  EventEmitter,
+  property,
+  querySlotAll,
+  returnOrFallthrough,
+} from '@clr/core/common';
+import {
+  checkCircleIcon,
+  ClarityIcons,
+  exclamationCircleIcon,
+  exclamationTriangleIcon,
+  infoCircleIcon,
+  timesIcon,
+} from '@clr/core/icon-shapes';
+import { html, LitElement } from 'lit-element';
+
+const iconMap = {
+  info: {
+    shape: ClarityIcons.getIconNameFromShape(infoCircleIcon),
+    title: CommonStringsService.keys.info,
+  },
+  success: {
+    shape: ClarityIcons.getIconNameFromShape(checkCircleIcon),
+    title: CommonStringsService.keys.success,
+  },
+  warning: {
+    shape: ClarityIcons.getIconNameFromShape(exclamationTriangleIcon),
+    title: CommonStringsService.keys.warning,
+  },
+  danger: {
+    shape: ClarityIcons.getIconNameFromShape(exclamationCircleIcon),
+    title: CommonStringsService.keys.danger,
+  },
+};
+
+/**
+ * Web component alerts.
+ *
+ * ```typescript
+ * import '@clr/core/alert';
+ * ```
+ *
+ * ```html
+ * <cwc-alert>
+ *    <cwc-alert-content>This is an alert.</cwc-alert-content>
+ * </cwc-alert>
+ * ```
+ *
+ * @noInheritDoc
+ * @beta 3.0
+ * @element cwc-alert
+ * @slot default - Content slot for inside the alert. Usually will contain at least a <cwc-alert-content> component.
+ */
+// @dynamic
```

Comment:
nit: probably should remove the tsdoc block from the base class since the child alert class will have its own definition.